### DISCUSSION TITLE
with opti: "enable_reverse" to false

### DISCUSSION THREAD

**Discussion by FaBiasch** [casadi/casadi#3934] [General]
Hi,

is there an option to disable reverse mode with something similar like `{"enable_reverse": false}` using `opti`? 

I also found a few issues, which say that this setting is not correctly propagated. Is this setting working?
Thanks.

**Comment by jaeandersson** [comment]
Try setting the 'common_options' dictionary when calling Opti.solver:
```python
opti.solver("ipopt", dict(common_options = dict(enable_reverse = False)))
```

@jgillis - correct me if I'm wrong

The "enable_reverse" option is not propagated when embedding functions. This is intentional. At each level, the framework will decide what kinds of derivatives to use. Often forward/reverse mode on the top level will result in the same type of derivative on lower levels, but not always. For example, if the number of directional derivatives exceeds the dimension of the embedded function, it will form a Jacobian and multiply by left or right. In your case, you probably want to disable reverse at the top level, i.e. NLP solver callback functions (jac_g and grad_f).
